hoplite-core/src/main/kotlin/com/sksamuel/hoplite/decoder/ranges.kt

```diff
@@ -0,0 +1,69 @@
+package com.sksamuel.hoplite.decoder
+
+import kotlin.reflect.KType
+
+import arrow.core.Try
+import arrow.core.getOrElse
+import arrow.data.invalid
+import arrow.data.valid
+import com.sksamuel.hoplite.ConfigFailure
+import com.sksamuel.hoplite.ConfigResult
+import com.sksamuel.hoplite.Node
+import com.sksamuel.hoplite.StringNode
+
+class IntRangeDecoder : NonNullableDecoder<IntRange> {
+  override fun supports(type: KType): Boolean = type.classifier == IntRange::class
+  override fun safeDecode(node: Node,
+                          type: KType,
+                          registry: DecoderRegistry): ConfigResult<IntRange> = when (node) {
+    is StringNode -> RangeDecoders.intRange(node.value).map { it.valid() }
+      .getOrElse { ConfigFailure.DecodeError(node, type).invalid() }
+    else -> ConfigFailure.DecodeError(node, type).invalid()
+  }
+}
+
+class LongRangeDecoder : NonNullableDecoder<LongRange> {
+  override fun supports(type: KType): Boolean = type.classifier == LongRange::class
+  override fun safeDecode(node: Node,
+                          type: KType,
+                          registry: DecoderRegistry): ConfigResult<LongRange> = when (node) {
+    is StringNode -> RangeDecoders.longRange(node.value).map { it.valid() }
+        .getOrElse { ConfigFailure.DecodeError(node, type).invalid() }
+    else -> ConfigFailure.DecodeError(node, type).invalid()
+  }
+}
+
+class CharRangeDecoder : NonNullableDecoder<CharRange> {
+  override fun supports(type: KType): Boolean = type.classifier == CharRange::class
+  override fun safeDecode(node: Node,
+                          type: KType,
+                          registry: DecoderRegistry): ConfigResult<CharRange> = when (node) {
+    is StringNode -> RangeDecoders.charRange(node.value).map { it.valid() }
+      .getOrElse { ConfigFailure.DecodeError(node, type).invalid() }
+    else -> ConfigFailure.DecodeError(node, type).invalid()
+  }
+}
+
+object RangeDecoders {
+
+  fun intRange(value: String): Try<IntRange> {
+    return extractContents(value, "(\\d+)\\.\\.(\\d+)".toRegex())
```

Comment:
I'm not really convinced about having the regex at this level, what do you think

Reply:
I'd pull it out into a val inside the object.